packages/zengine-sdk/src/filter-matcher.ts

```diff
@@ -0,0 +1,263 @@
+    if (Array.isArray(recordValue)) {
+      return recordValue.length > 0
+    }
+    return ruleValue.toString().toLowerCase() !== recordValue.toString().toLowerCase()
+  },
+  ruleMinimum: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+      return false
+    }
+    if (recordValue === '') {
+      return false
+    }
+    // Do numeric comparison if filter rule value is numeric. For consistency with API,
+    // record values are somewhat forgiving, while filter rules must be a valid number.
+    if (ruleValue && !isNaN(Number(ruleValue))) {
+      recordValue = parseNumber(recordValue)
+
+      try {
+        ruleValue = parseFloat(ruleValue)
+        recordValue = parseFloat(recordValue)
+      } catch (err) {
+        return false
+      }
+
+      return recordValue >= ruleValue
+    }
+    // Loose comparison, don't use 'localeCompare' because we only want to compare
+    // numbers or well-formatted dates, which work with normal string comparison.
+    return recordValue >= ruleValue
+  },
+  ruleMaximum: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+      return false
+    }
+    if (recordValue === '') {
+      return false
+    }
+    // Do numeric comparison if filter rule value is numeric. For consistency with API,
+    // record values are somewhat forgiving, while filter rules must be a valid number.
+    if (ruleValue && !isNaN(Number(ruleValue))) {
+      recordValue = parseNumber(recordValue)
+
+      try {
+        ruleValue = parseFloat(ruleValue)
+        recordValue = parseFloat(recordValue)
+      } catch (err) {
+        return false
+      }
+
+      return recordValue <= ruleValue
+
+    }
+    // Loose comparison, don't use 'localeCompare' because we only want to compare
+    // numbers or well-formatted dates, which work with normal string comparison.
+    return recordValue <= ruleValue
+  },
+  ruleContains: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+
+      // Normalize Value as Array
+      if (!Array.isArray(ruleValue)) {
+        ruleValue = [ruleValue]
+      }
+
+      for (let index = 0; index < ruleValue.length; index++) {
+        if (recordValue.indexOf(ruleValue[index]) === -1) {
+          return false
+        }
+      }
+      return true
+    } else {
+      return recordValue.toString().indexOf(ruleValue.toString()) !== -1
+    }
+  },
+  ruleDoesNotContain: function (recordValue: any, ruleValue: any) {
+    return !matchers.ruleContains(recordValue, ruleValue)
+  },
+  ruleStartsWith: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+      return false
+    }
+    return recordValue.toString().startsWith(ruleValue.toString())
+  },
+  ruleEndsWith: function (recordValue: any, ruleValue: any) {
+    if (Array.isArray(recordValue)) {
+      return false
+    }
+    return recordValue.toString().endsWith(ruleValue.toString())
```

Comment:
what confidence do we have that startsWith/endsWith doesn't need a polyfill? I think TSLib inserts those, but we should definitely double check, since we technically support IE.

Reply:
supported in Edge not IE <= 11, checking tslib, I guess anglerfish adds a polyfill for them